File: CS-Examples/10_Conversion/HtmlStringToWord.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using System.IO;

namespace HtmlStringToWord
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Read the content of an HTML file into a string variable.
			string HTML = File.ReadAllText(@"..\..\..\..\..\..\..\Data\InputHtml.txt");

			// Create a new instance of the Document class.
			Document document = new Document();

			// Add a new section to the document.
			Section sec = document.AddSection();

			// Add a new paragraph to the section and append the HTML content to it.
			sec.AddParagraph().AppendHTML(HTML);

			// Save the document as a DOCX file named "HtmlFileToWord.docx".
			document.SaveToFile("HtmlFileToWord.docx", FileFormat.Docx);

			// Dispose the document object to release resources.
			document.Dispose();

            //Launch the Word file.
            WordDocViewer("HtmlFileToWord.docx");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/10_Conversion/HtmlStringToWord.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc
Imports System.IO

Namespace HtmlStringToWord
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Read the contents of the HTML file into a string variable
			Dim HTML As String = File.ReadAllText("..\..\..\..\..\..\..\Data\InputHtml.txt")

			' Create a new Document object
			Dim document As New Document()

			' Add a new section to the document
			Dim sec As Section = document.AddSection()

			' Add a paragraph to the section and append the HTML content to it
			sec.AddParagraph().AppendHTML(HTML)

			' Save the document to a Word file in Docx format with the specified output file name
			document.SaveToFile("HtmlFileToWord.docx", FileFormat.Docx)

			' Dispose of the Document object to release resources
			document.Dispose()

			'Launch the Word file.
			WordDocViewer("HtmlFileToWord.docx")
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
